tcp-native/src/main/java/org/ballerinalang/stdlib/tcp/TcpClient.java

```diff
@@ -53,18 +67,54 @@ public TcpClient(InetSocketAddress localAddress, InetSocketAddress remoteAddress
                     @Override
                     protected void initChannel(SocketChannel ch) throws Exception {
                         ch.pipeline().addLast(Constants.CLIENT_HANDLER, new TcpClientHandler());
+                        if (secureSocket != null) {
+                            setSSLHandler(ch, secureSocket);
+                        }
                     }
-                });
-        ChannelFuture future = clientBootstrap.connect(remoteAddress, localAddress).sync()
+
+                    @Override
+                    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+                        callback.complete(Utils.createSocketError(cause.getMessage()));
+                        ctx.close();
+                    }
+                })
+                .connect(remoteAddress, localAddress).sync()
                 .addListener((ChannelFutureListener) channelFuture -> {
-            if (channelFuture.isSuccess()) {
-                channelFuture.channel().config().setAutoRead(false);
-                channel = channelFuture.channel();
-                callback.complete(null);
-            } else {
-                callback.complete(Utils.createSocketError("Unable to connect with remote host."));
-            }
-        });
+                    if (channelFuture.isSuccess()) {
+                        channelFuture.channel().config().setAutoRead(false);
+                        channel = channelFuture.channel();
+                        callback.complete(null);
+                    } else {
+                        callback.complete(Utils.createSocketError("Unable to connect with remote host."));
+                    }
+                });
+    }
+
+    private void setSSLHandler(SocketChannel ch, BMap<BString, Object> secureSocket) throws NoSuchAlgorithmException,
```

Comment:
Using complete names for variable names is our convention.  Shall we use `channel` instead of `ch`?

Reply:
updated